examples/mne_connectivity_classification.py

```diff
@@ -0,0 +1,202 @@
+# that comprises of the different sleep stage events with 30 seconds
+# of data in each Epoch.
+
+tmax = 30. - 1. / raw_train.info['sfreq']  # tmax in included
+
+epochs_train = mne.Epochs(raw=raw_train, events=events_train,
+                          event_id=event_id, tmin=0., tmax=tmax, baseline=None)
+times = epochs_train.times
+ch_names = raw_train.ch_names
+
+epochs_train.drop_bad()
+
+print(epochs_train)
+
+##############################################################################
+# Applying the same steps to the test data from Bob
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+raw_test = mne.io.read_raw_edf(bob_files[0])
+annot_test = mne.read_annotations(bob_files[1])
+annot_test.crop(annot_test[1]['onset'] - 30 * 60,
+                annot_test[-2]['onset'] + 30 * 60)
+raw_test.set_annotations(annot_test, emit_warning=False)
+raw_test.set_channel_types(mapping)
+
+# Now we band-pass filter our data.
+raw_test.load_data()
+raw_test.filter(14, 30)
+
+events_test, _ = mne.events_from_annotations(
+    raw_test, event_id=annotation_desc_2_event_id, chunk_duration=30.)
+epochs_test = mne.Epochs(raw=raw_test, events=events_test, event_id=event_id,
+                         tmin=0., tmax=tmax, baseline=None)
+
+epochs_test.drop_bad()
+
+print(epochs_test)
+
+##############################################################################
+# Compute classification model
+# ----------------------------
+# Here, we use :func:`mne_connectivity.envelope_correlation` and a
+# ``RandomForestClassifier`` trained
+# on the envelope correlation values to classify sleep stage. We leverage
+# scikit-learn's ``FunctionTransformer`` to create a ``Pipeline`` object
+# for classification.
+
+
+def apply_degree_across_epochs(conn, **kwargs):
+    """Apply degree across epochs and aggregate."""
+    degrees = []
+    epoch_conn = conn.get_data(output='dense')
+    for epoch_idx in range(conn.n_epochs):
+        conn_data = epoch_conn[epoch_idx, ...].squeeze()
+        degrees.append(degree(conn_data, **kwargs))
+    return np.array(degrees)
+
+kw_args = dict(names=ch_names,)
+
+
+# create sklearn pipeline
+pipe = make_pipeline(
+    FunctionTransformer(envelope_correlation, kw_args=kw_args),
+    FunctionTransformer(lambda x: x.get_data(output='raveled').squeeze()),
+    # FunctionTransformer(apply_degree_across_epochs,
+    # kw_args=dict(threshold_prop=0.01)),
+    RandomForestClassifier(n_estimators=100, random_state=42),
+)
+
+# Train
+y_train = epochs_train.events[:, 2]
+pipe.fit(epochs_train, y_train)
+
+# Test
+y_pred = pipe.predict(epochs_test)
+
+# Assess the results
+y_test = epochs_test.events[:, 2]
+acc = accuracy_score(y_test, y_pred)
+
+# dummy classifier
+dummby_clf = DummyClassifier(strategy='most_frequent')
+dummby_clf.fit(epochs_train, y_train)
+dummby_pred = dummby_clf.predict(epochs_test)
+dummby_acc = accuracy_score(y_test, dummby_pred)
+
+# We perform slightly better then chance as an initial first pass.
```

Comment:
Well. This is most likely not significant and we probably should not promote this (compare to `DummyClassifier` by just looking at the scores) as a way of doing classification without doing a proper significant test.